frontend/src/entities/configuration/dynamic-configuration.tsx

```diff
@@ -0,0 +1,123 @@
+/**
+  Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+
+  Licensed under the Apache License, Version 2.0 (the "License").
+  You may not use this file except in compliance with the License.
+  You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+*/
+import {
+    ExpandableSection,
+    Header,
+    SpaceBetween,
+    Container,
+    Button,
+    Input,
+} from '@cloudscape-design/components';
+import React, { useEffect } from 'react';
+import { useAppDispatch, useAppSelector } from '../../config/store';
+import { appConfig, appConfigList, getConfiguration, listConfigurations, updateConfiguration } from './configuration-reducer';
+import { IAppConfiguration } from '../../shared/model/app.configuration.model';
+import { useValidationReducer } from '../../shared/validation';
+import { z } from 'zod';
+
+export function DynamicConfiguration () {
+    const applicationConfig: IAppConfiguration = useAppSelector(appConfig);
+    const configList: IAppConfiguration[] = useAppSelector(appConfigList);
+
+    const formSchema = z.object({//configuration.EMRConfig.clusterSizes[0].name
+        configuration: z.object({
+            EMRConfig: z.object({
+                clusterSizes: z.object({
+                    name: z.string({ required_error: 'A cluster name must be provided.' })
+                })
+            })
+        }),
+    });
+
+    const { state, setState, setFields, touchFields } = useValidationReducer(formSchema, {
+        validateAll: false as boolean,
+        needsValidation: false,
+        touched: {},
+        form: {
+            ...applicationConfig,
+        },
+        formValid: false,
+        formSubmitting: false as boolean,
+    });
+
+    const dispatch = useAppDispatch();
+    useEffect(() => {
+        dispatch(listConfigurations({configScope: 'global', numVersions: 5}));
+    }, [dispatch]);
+
+    const handleSubmit = async () => {
```

Comment:
I think you should be able to avoid that blurb by using the `isValid` result from `useValidationReducer`. As of recently [it is returned](https://github.com/awslabs/mlspace/blob/43b1253b2b865bcaa73d44c255a9fd550c1d3b23/frontend/src/shared/validation/index.ts#L296-L308) so we can avoid validating forms twice.

Reply:
I'll try using the `isValid` value, let me know if I use it incorrectly

Reply:
Also added notifications alerting the user to:

1. new config updated successfully
2. new config update failed due to a 429 error (you're operating on an outdated config)
3. new config update failed for an unknown reason, generic error message